java/src/main/java/io/cucumber/gherkin/GherkinDocumentBuilder.java

```diff
@@ -245,32 +245,41 @@ private static String joinMatchedText(List<Token> lineTokens, int endIndex) {
         return content.toString();
     }
 
+    @SuppressWarnings("ForLoopReplaceableByForEach") // classic 'for' loop is ~2x faster than 'for-each'
     private List<TableRow> getTableRows(AstNode node) {
         List<Token> tokens = node.getTokens(TokenType.TableRow);
-        List<TableRow> rows = new ArrayList<>(tokens.size());
+        int tokenCount = tokens.size();
```

Comment:
Nitpick: 
 * count -> a number obtained by iterating over some array or list
 * size ->  a property of a collection
 * length -> a property of an array.
 
 So I think using `tokenSize` here would be the correct choice.